exporters/jaeger/lib/opentelemetry/exporters/jaeger/exporter.rb

```diff
@@ -0,0 +1,174 @@
+        # this exporter is registered to a {OpenTelemetry::SDK::Trace::Tracer}
+        # object.
+        def shutdown
+          @shutdown = true
+        end
+
+        private
+
+        def batcher
+          batch = 0
+          batch_size = 0
+          ->(_, span_size) {
+            if batch_size + span_size > @max_packet_size
+              batch_size = 0
+              batch += 1
+            end
+            batch_size += span_size
+            batch
+          }
+        end
+
+        # Yields Thrift-encoded batches of spans. Batches are limited to @max_packet_size.
+        # If a single span exceeds @max_packet_size, FAILED_NOT_RETRYABLE will be returned
+        # and the remaining batches will be discarded. Returns SUCCESS after all batches
+        # have been successfully yielded.
+        def encode_batches(span_data)
+          encoded_spans = span_data.map(&method(:encode_span))
+          encoded_span_sizes = encoded_spans.map(&method(:encoded_span_size))
+          return FAILED_NOT_RETRYABLE if encoded_span_sizes.any? { |size| size > @max_packet_size }
+
+          encoded_spans.zip(encoded_span_sizes).chunk(&batcher).each do |array|
+            yield Thrift::Batch.new('process' => encode_process, 'spans' => array.map(&:last))
+          end
+          SUCCESS
+        end
+  
+        def encode_span(span_data)
+          start_time = (span_data.start_timestamp.to_f * 1_000_000).to_i
+          duration = (span_data.end_timestamp.to_f * 1_000_000).to_i - start_time
+  
+          Thrift::Span.new(
+            'traceIdLow' => int64(span_data.trace_id[16, 16]),
+            'traceIdHigh' => int64(span_data.trace_id[0, 16]),
+            'spanId' => int64(span_data.span_id),
+            'parentSpanId' => int64(span_data.parent_span_id),
+            'operationName' => span.name,
+            'references' => encode_references(span_data.links),
+            'flags' => span_data.trace_flags.sampled? ? 1 : 0,
+            'startTime' => start_time,
+            'duration' => duration,
+            'tags' => encode_tags(span_data.attributes),
+            'logs' => encode_logs(span_data.events)
+          )
+        end
+
+        def encode_logs(events)
+          events.map do |event|
+            Thrift::Log.new(
+              'timestamp' => (event.timestamp.to_f * 1_000_000).to_i,
+              'fields' => encode_tags(event.attributes) + encode_tags('name' => event.name)
+            )
+          end
+        end
+
+        def encode_references(links)
+          links.map do |link|
+            Thrift::SpanRef.new(
+              'refType' => Thrift::SpanRefType::CHILD_OF,
+              'traceIdLow' => int64(link.context.trace_id[16, 16]),
+              'traceIdHigh' => int64(link.context.trace_id[0, 16]),
+              'spanId' => int64(link.context.span_id)
+            )
+          end
+        end
+
+        def encode_tags(attributes)
+          @type_map ||= {
```

Comment:
you could also do something like this https://github.com/salemove/jaeger-client-ruby/blob/6ecc35fcc79153625fd9a8d6e563f30d5a4f594b/lib/jaeger/thrift_tag_builder.rb#L4 to get some compile time checking of keys and to avoid using strings directly

Reply:
Addressed in https://github.com/open-telemetry/opentelemetry-ruby/pull/112/commits/500c49065de98f0818c476102e8e9acbaba5c503